nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/io/download/TestDownloadService.java

```diff
@@ -535,4 +534,11 @@ public void write(int b) throws IOException {
         assertEquals(blobValue, out.toString()); // decode with system default
     }
 
+    @Test
+    public void testDownloadUrlWithJSessionIDSanitization() throws IOException {
+        String filename = "/home/john/foo.txt;jsessionid=FooBarBaz";
```

Comment:
Please also test with a URL with another matrix parameter, and a query parameter, like:

    "/home/john/foo.txt;jsessionid=FooBarBaz;otherparam=foobarbaz?query=value"
  
Also test with a URL that does *not* have the `;jsessionid=xyz` part

Reply:
the `?query=value` is handled correctly by the DownloadServlet, only the `;` is not properly handled